pxr/usdValidation/usdShadeValidators/testenv/testUsdShadeValidators.cpp

```diff
@@ -554,13 +558,223 @@ TestUsdShadeEncapsulationRulesValidator()
     }
 }
 
+void ValidateError(const UsdValidationErrorVector& errors,
+        const TfToken& expectedErrorIdentifier,
+        const SdfPath& expectedPrimPath,
+        const std::string& expectedErrorMsg,
+        UsdValidationErrorType expectedErrorType =
+        UsdValidationErrorType::Error)
+{
+    TF_AXIOM(errors.size() == 1);
+    TF_AXIOM(errors[0].GetIdentifier() == expectedErrorIdentifier);
+    TF_AXIOM(errors[0].GetType() == expectedErrorType);
+    TF_AXIOM(errors[0].GetSites().size() == 1);
+    TF_AXIOM(errors[0].GetSites()[0].IsValid());
+    TF_AXIOM(errors[0].GetSites()[0].IsPrim());
+    TF_AXIOM(errors[0].GetSites()[0].GetPrim().GetPath() ==
+             expectedPrimPath);
+    TF_AXIOM(errors[0].GetMessage() == expectedErrorMsg);
+}
```

Comment:
Thanks for this; Might make sense to also update other testing methods here to use this, could you do that also, as you are introducing this common method?

Reply:
Happy to do this. Does it make sense to bring this helper function up to `usdValidation` and then reuse it in all domain specific validators?

If so, do you want that on this PR or on a separate PR?

Reply:
Hmm good thought, I feel based on other unit tests in all of openUsd, a pattern established is to have helper like these stick with each unit test. So I think its fine to have this helper function for each test file.